### PR TITLE
Prepare Stargate package for CosmWasm support

### LINKED ISSUE #376: Create a log parser for Stargate client
I.e. for parsing raw logs in `.broadcastTx()` results.

### PR DESCRIPTION
**Pull request by willclarktech**
Part 1 of #491
Closes #376

### CODE REVIEW

File: packages/stargate/scripts/define-proto.sh [hunk 1]
```diff
@@ -1,13 +1,12 @@
...
 set -o errexit -o nounset -o pipefail
-command -v shellcheck > /dev/null && shellcheck "$0"
+command -v shellcheck >/dev/null && shellcheck "$0"
 
 TMP_DIR="./tmp"
 JS_SOURCE_FILE="$TMP_DIR/codecimpl.js"
 DEFINITIONS_FILE="$TMP_DIR/codecimpl.d.ts"
 OUTPUT_DIR="./src/codec/generated/"
 
-
```
Comment (webmaster128) [inline packages/stargate/scripts/define-proto.sh:10]: Whiy this diff? This is already done in https://github.com/cosmos/cosmjs/pull/567

File: packages/stargate/src/queries/queryclient.ts [hunk 2]
```diff
@@ -234,7 +234,7 @@ export class QueryClient {
   // throws an error if height is 0 or undefined
   private async getNextHeader(height?: number): Promise<Header> {
     assert(height);
-    if (height == 0) {
+    if (height === 0) {
```
Comment (webmaster128) [inline packages/stargate/src/queries/queryclient.ts:237]: The `assert` already guarantees this is not 0. We could use `assertDefined(height);` instead to keep the specific error message for 0.

File: packages/stargate/src/stargateclient.ts [hunk 3]
```diff
@@ -54,14 +60,14 @@ export interface BroadcastTxFailure {
   readonly code: number;
   readonly transactionHash: string;
   readonly rawLog?: string;
-  readonly data?: Uint8Array;
+  readonly data?: readonly IMsgData[];
```
Comment (webmaster128) [inline packages/stargate/src/stargateclient.ts:63]: Ideally we has this, such that there is always one entry per message. But I see that the protobuf type system makes this hard to get.

```suggestion
  readonly data: readonly IMsgData[];
```

**Review by webmaster128 — COMMENTED** [review]
Does this close https://github.com/cosmos/cosmjs/issues/376?

File: packages/stargate/scripts/define-proto.sh (hunk 1 above)
Comment (willclarktech) [reply packages/stargate/scripts/define-proto.sh:10]: Ah, need to rebase

**Review by webmaster128 — APPROVED** [review]

**Comment by willclarktech** [comment]
> Does this close #376?

I think so!

**Pull request merged**